### ISSUE TITLE
CSS module scripts

### ISSUE THREAD

**Issue by uasan** [WebKit/standards-positions#77]
## Request for position on an emerging web specification

 Implement CSS module scripts

* WebKittens who can provide input: 

### Information about the spec

* Spec Title: CSS module scripts
* Spec URL:  https://html.spec.whatwg.org/multipage/webappapis.html#creating-a-css-module-script
* GitHub repository: https://github.com/whatwg/html/pull/4898
* Issue Tracker (if not the repository's issue tracker): 
* Explainer (if not README.md in the repository): https://github.com/WICG/webcomponents/blob/gh-pages/proposals/css-modules-v1-explainer.md

### Design reviews and vendor positions

* TAG Design Review: 
* Mozilla standards-positions issue: 

### Bugs tracking this feature

* WebKit Bugzilla: https://bugs.webkit.org/show_bug.cgi?id=227967
* Radar: 

### Anything else we need to know

The CSS module scripts feature has been merged into the HTML spec.

**Comment by annevk** [comment]
Adding `concerns: integration` due to https://github.com/whatwg/html/issues/7233 still existing.

**Comment by annevk** [comment]
Note that as per references in #128 the underlying syntax is likely to change here.

**Comment by clshortfuse** [comment]
From a developer perspective I'm transcompiling imports into a constructable `CSSStyleSheet` for Firefox. 

On Safari (non TP), I put it into a `HTMLStyleSheet` and pass the child `CSSStyleSheet` that's still attached. Before adding a CSSStyleSheet to a ShadowDOM (Web Components), the `CSSStyleSheet` is inspected and the parent `HTMLStyleSheet` gets cloned. I imagine if/when constructable `CSSStyleSheet`s is out ot TP, I'll be able to reduce some code, but still would need transcompilation.  

I hope it gets implemented, but not entirely important for production since we package imports before deployment. For development, we'd have 0 transcompile which means we can work directly against files on the drive instead of minified (`FormAssociated` is/was our only polyfill). That can be rather useful for live-editing styles within the browser (easier than source maps). We do not expect to move back to SCSS, or author CSS within JS.

**Comment by annevk** [comment]
This would be "position: support" if https://github.com/WebKit/standards-positions/issues/77#issuecomment-1290347676 were addressed. Marking this blocked for now.

**Comment by uasan** [comment]
As I understand it, this commit should unblock this task?
https://github.com/WebKit/WebKit/commit/35c672ab42641db0295b649a694a3355bf2b03e2

**Comment by annevk** [comment]
As https://github.com/WebKit/standards-positions/issues/77#issuecomment-1290347676 has been resolved (thanks Nicolò!) and no other concerns have been raised I suggest we mark this "position: support" one week from now.
